pandas/core/arrays/period.py

```diff
@@ -587,6 +587,13 @@ def astype(self, dtype, copy: bool = True):
             return self.asfreq(dtype.freq)
         return super().astype(dtype, copy=copy)
 
+    def searchsorted(self, value, side="left", sorter=None):
+        value = self._validate_searchsorted_value(value).view("M8[ns]")
+
+        # Cast to M8 to get datetime-like NaT placement
+        m8arr = self._ndarray.view("M8[ns]")
```

Comment:
aren't these ordinals?

Reply:
Yes.  thats why this is viewing as dt64 in order to get searchsorted NaT placement to match dt64/td64